internal/usecases/watcher/eclipse_check_test.go

```diff
@@ -0,0 +1,581 @@
+		useCase := w.NewEclipseCheckUseCase(
+			config,
+			blockchain.Rpc{
+				Btc: btcRpc,
+				Rsk: rskRpc,
+			},
+			[]blockchain.BitcoinNetwork{},
+			[]blockchain.RootstockRpcServer{rskExtra1, rskExtra2},
+			eventBus,
+			alertSender,
+			recipient,
+			mutex,
+		)
+		err := useCase.Run(context.Background(), entities.NodeTypeRootstock)
+		require.NoError(t, err)
+		rskRpc.AssertExpectations(t)
+		for _, rskExtra := range rskExtras {
+			rskExtra.AssertExpectations(t)
+		}
+		eventBus.AssertNotCalled(t, "Publish")
+		alertSender.AssertNotCalled(t, "SendAlert")
+		mutex.AssertExpectations(t)
+	})
+}
+
+// nolint:funlen
+func TestEclipseCheckUseCase_Run_Bitcoin(t *testing.T) {
+	const (
+		recipient   = "alertRecipient"
+		networkName = "testnet"
+		blocks      = 500
+		headers     = 501
+	)
+	config := w.EclipseCheckConfig{
+		RskToleranceThreshold:    0,
+		RskMaxMsWaitForBlock:     0,
+		RskWaitPollingMsInterval: 0,
+		BtcToleranceThreshold:    50,
+		BtcMaxMsWaitForBlock:     1000,
+		BtcWaitPollingMsInterval: 100,
+	}
+	t.Run("should not trigger the alert if no eclipse attack is detected", func(t *testing.T) {
+		rskRpc := &mocks.RootstockRpcServerMock{}
+		btcRpc := &mocks.BtcRpcMock{}
+		btcExtra1 := &mocks.BtcRpcMock{}
+		btcExtra2 := &mocks.BtcRpcMock{}
+		eventBus := &mocks.EventBusMock{}
+		alertSender := &mocks.AlertSenderMock{}
+		btcExtras := []*mocks.BtcRpcMock{btcExtra1, btcExtra2}
+		btcRpc.On("GetBlockchainInfo").Return(blockchain.BitcoinBlockchainInfo{
+			NetworkName:      networkName,
+			ValidatedBlocks:  big.NewInt(blocks),
+			ValidatedHeaders: big.NewInt(headers),
+			BestBlockHash:    test.AnyHash,
+		}, nil)
+		// 50% should be enough to not trigger the alert
+		btcExtras[0].On("GetBlockchainInfo").Return(blockchain.BitcoinBlockchainInfo{
+			NetworkName:      networkName,
+			ValidatedBlocks:  big.NewInt(blocks),
+			ValidatedHeaders: big.NewInt(headers),
+			BestBlockHash:    test.AnyHash,
+		}, nil)
+		btcExtras[1].On("GetBlockchainInfo").Return(blockchain.BitcoinBlockchainInfo{
+			NetworkName:      networkName,
+			ValidatedBlocks:  big.NewInt(100),
+			ValidatedHeaders: big.NewInt(101),
+			BestBlockHash:    "otherHash",
+		}, nil)
+		mutex := &mocks.MutexMock{}
+		mutex.On("Lock").Return()
+		mutex.On("Unlock").Return()
+		useCase := w.NewEclipseCheckUseCase(
+			config,
+			blockchain.Rpc{
+				Btc: btcRpc,
+				Rsk: rskRpc,
+			},
+			[]blockchain.BitcoinNetwork{btcExtras[0], btcExtras[1]},
+			[]blockchain.RootstockRpcServer{},
+			eventBus,
+			alertSender,
+			recipient,
+			mutex,
+		)
+		err := useCase.Run(context.Background(), entities.NodeTypeBitcoin)
+		require.NoError(t, err)
+		btcRpc.AssertExpectations(t)
+		for _, btcExtra := range btcExtras {
+			btcExtra.AssertExpectations(t)
+		}
+		eventBus.AssertNotCalled(t, "Publish")
+		alertSender.AssertNotCalled(t, "SendAlert")
+		mutex.AssertExpectations(t)
+	})
+	t.Run("should trigger the alert if eclipse attack is detected", func(t *testing.T) {
```

Comment:
This test also use the explicit values of blocks and headers instead of the constants